addons/web/static/src/views/list/list_renderer.js

```diff
@@ -1478,6 +1509,193 @@ export class ListRenderer extends Component {
+     * @returns {iterable} containing definitions as fields {fieldName: [propertyValues]}
+     */
+    getPropertyDefinitions(props, fieldName, parentModel) {
+        const propertiesData = {};
+        for (const record of props.list.records) {
+            const properties = record.data[fieldName] || [];
+            for (const property of properties) {
+                if (!(parentModel in record.data)) {
+                    //TODO rpc + async update instead of this?
+                    throw new Error(`A custom property field ${fieldName} requires its parent ${parentModel} field to be in the view as well`);
+                }
+                propertiesData[this.getPropertyFieldName(property.name, fieldName)] = {
+                    ...property,
+                    parentId: record.data[parentModel][0],
+                    parentName: record.data[parentModel][1],
+                };
+            }
+        }
+        return Object.values(propertiesData).flat();
+    }
+
+    getNonPropertyColumns(columns) {
+        return columns.filter(column => !this.getPropertyParentModel(column.name));
+    }
+
+    /**
+     * Group property columns in a structure that allows to display them based on where they are defined.
+     * The structure is such that it can easily be iterated over in a forEach loop.
+     *
+     * @example [column1, ...] -> [{definitionModel: 'team_id', group: [{parentId: 3, columns: [column1, ...]}, ...]}, ...]
+     * @param {object} columns virtual property columns generated via ``getVirtualPropertyFields``
+     * @returns {Array} inlined groups per definition model, then per specific definition record
+     */
+    getPropertyColumnGroups(columns) {
+        const propertyColumns = columns.filter(column => this.getPropertyParentModel(column.name));
+        const grouped = {};
+        for (const column of propertyColumns) {
+            const columnPropertyDefinition = this.getPropertyParentModel(column.name);
+
+            if (!grouped[columnPropertyDefinition]) {
+                grouped[columnPropertyDefinition] = {};
+            }
+            if (!grouped[columnPropertyDefinition][column.propertyParentId]) {
+                grouped[columnPropertyDefinition][column.propertyParentId] = [];
+            }
+
+            grouped[columnPropertyDefinition][column.propertyParentId].push(column);
+        }
+
+        const inlinedDefinitionGroups = [];
+        for (const [definition, definitionGroup] of Object.entries(grouped)) {
+            const inlinedGroup = [];
+            for (const [parent, columns] of Object.entries(definitionGroup)) {
+                // if there's an entry in the group, there's at least one column in it
+                inlinedGroup.push({parentId: parent, parentName: columns[0].propertyParentName, columns: columns});
+            }
+            inlinedDefinitionGroups.push({definitionModel: definition, group: inlinedGroup});
+        }
+        return inlinedDefinitionGroups;
```

Comment:
I think it's possible to do 2 nested array.map here (for `inlinedDefinitionGroups` and for `inlinedGroup`)